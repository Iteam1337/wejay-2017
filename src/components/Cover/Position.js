import React, { Component } from 'react'

class Position extends Component {
  state = {
    position: 1500
  }

  timer = null

  componentDidMount () {
    this.startTimer(this.props)
  }

  componentWillReceiveProps (nextProps) {
    console.log(this.props.track, nextProps.track)
    if (this.props.track.spotifyUri !== nextProps.track.spotifyUri) {
      clearInterval(this.timer)

      this.startTimer(nextProps)
    }
  }

  startTimer = props => {
    const { track } = props

    this.timer = setInterval(() => {
      this.setState(() => ({
        position: 1500 - (Date.now() - track.started) / track.duration * 1500
      }))
    }, 1000)
  }

  render () {
    const { track } = this.props
    const style = {
      height: 310 / track.album.images[0].width * track.album.images[0].height,
      width: 310 / track.album.images[0].width * track.album.images[0].width
    }

    return (
      <div className="Position" style={style}>
        <svg
          height="310"
          width="310"
          xmlns="http://www.w3.org/2000/svg"
          style={style}
        >
          <rect
            className="shape"
            height="310"
            width="310"
            style={{
              ...style,
              strokeDashoffset: this.state.position
            }}
          />
        </svg>
      </div>
    )
  }
}

export default Position
