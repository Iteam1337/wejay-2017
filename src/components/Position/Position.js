// @flow

import React, { Component } from 'react'
import type { Track } from 'models/models'

type Props = {
  track: Track
}

type State = {
  position: number
}

class PositionTracker extends Component<Props, State> {
  state = {
    position: 0,
  }

  timer = undefined

  componentDidMount () {
    this.startTimer(this.props)
  }

  componentWillReceiveProps (nextProps: Props) {
    if (this.props.track.spotifyUri !== nextProps.track.spotifyUri) {
      clearInterval(this.timer)

      this.startTimer(nextProps)
    }
  }

  componentWillUnmount () {
    clearInterval(this.timer)
  }

  startTimer = (props: Props) => {
    const { track } = props

    this.timer = setInterval(() => {
      this.setState(() => ({
        position: (Date.now() - track.started) / track.duration * 100,
      }))
    }, 1000)
  }

  render () {
    const { position } = this.state

    return (
      <div className="Position">
        <div className="Position__bar" style={{ width: `${position}%` }} />
      </div>
    )
  }
}

export default PositionTracker
