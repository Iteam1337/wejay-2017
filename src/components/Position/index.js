import PositionTracker from './Position'
export default PositionTracker
