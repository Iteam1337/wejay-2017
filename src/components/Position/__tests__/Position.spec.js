import React from 'react'
import { shallow } from 'enzyme'
import Position from '../Position'

describe('components/Position', () => {
  let component

  beforeEach(() => {
    component = shallow(<Position />)
  })

  it('renders Position', () => {
    expect(component).toMatchSnapshot()
  })
})
