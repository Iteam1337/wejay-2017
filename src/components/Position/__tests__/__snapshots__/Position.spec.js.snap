// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Position renders Position 1`] = `
<div
  className="Position"
>
  <div
    className="Position__bar"
    style={
      Object {
        "width": "0%",
      }
    }
  />
</div>
`;
